### PR TITLE
Created Gamma distribution class and added it to init

### PR DESCRIPTION
**Pull request by The-Shwin**
PR to solve issue #15 (https://github.com/hot9cups/probdists/issues/15)

I created a Gamma distribution class and added it to init.py. Please let me know if there's anything that needs to be adjusted. Thanks!

### CODE REVIEW

**Review by hot9cups — COMMENTED** [review]
@The-Shwin Thanks for taking the time to contribute. I'm quite happy with most of the code. One thing you'll have to resolve are the issues that the checks are showing up above.

Secondly, could you add some tests in [this file](https://github.com/hot9cups/probdists/blob/main/test.py) and check if everything works fine? 

I would also like you to add a data file for this distribution, a txt file with 10 numbers.

Finally, are you also interested in getting your hands dirty with some documentation and updating your work in [usage.md](https://github.com/hot9cups/probdists/blob/main/USAGE.md) ?

**Comment by The-Shwin** [comment]
> 
> 
> @The-Shwin Thanks for taking the time to contribute. I'm quite happy with most of the code. One thing you'll have to resolve are the issues that the checks are showing up above.
> 
> Secondly, could you add some tests in [this file](https://github.com/hot9cups/probdists/blob/main/test.py) and check if everything works fine?
> 
> I would also like you to add a data file for this distribution, a txt file with 10 numbers.
> 
> Finally, are you also interested in getting your hands dirty with some documentation and updating your work in [usage.md](https://github.com/hot9cups/probdists/blob/main/USAGE.md) ?

I have everything but the Codacy quality review covered. The issue with unused imports in __init__.py cannot be resolved if we are to follow the convention of adding those imports there for someone else to use in their code via init. The remaining linter error contradicts itself when I attempt to correct, so I cannot resolve it. It simply reverses state (blank line required-> no blank lines allowed and vice-versa). This was the fewest Codacy errors I could get.

File: probdists/Gammadistribution.py [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+    def __init__(self, k=2, theta=2, fit=False, data_file='demo_gamma_data'):
+        """
+        Init function to instantiate Gamma distribution
+            Args:
+                k (float) shape parameter representing shape of distribution (k > 0)
+                theta (float) scale parameter that stretches/shrinks distribution (theta > 0)
+        """
+        if k <= 0 or theta <= 0:
+            raise ValueError
+        if fit is False:
```
Comment (hot9cups) [inline probdists/Gammadistribution.py:26]: I think PEP8 recommends `if not fit:`

File: USAGE.md [hunk 2]
```diff
@@ -122,3 +122,42 @@
...
+
+# By default Gamma will create a gamma dist. with k=2, theta=2 and won't fit to data
+# This is useful if you want a simple Gamma distribution to play around with the pdf
+# HOWEVER: It will not make use of data passed into the distribution
+>>> gamma = Gamma()
+
+# To use of the sample data or your own data, and approximate a gamma fit to that data:
+>>> gamma = Gamma(fit=True, data_file='demo_gamma_data')
+# for your own file, replace 'demo_gamma_data' with 'my_data_file.txt' 
+# Ensure there is no extra whitespace at end of file
```
Comment (hot9cups) [inline USAGE.md:138]: This is being taken care of in another PR, where the read_data_file() function is being improved to be more robust and deal with such cases

File: USAGE.md [hunk 3]
```diff
@@ -122,3 +122,42 @@
...
+>>> print(gamma.pdf(x))
+0.18165
+
+# to access data 
+>>> print(gamma.data)
+[1, 2, 2, 3, 3, 4, 5, 6, 8, 9, 13]
+
+# plot pdf of exponential distribution 
+>>> gamma.plot_bar_pdf() 
+```
```
Comment (hot9cups) [inline USAGE.md:163]: Everything else looks fine here, except that you've coded up the `__add__()` magic method, and haven't documented here. Could you go ahead and do that as well?

**Review by hot9cups — COMMENTED** [review]
@The-Shwin mm, well for the unused import, you could add `# NOQA: F401` to that line which helps resolve the issue.
However, I've gone ahead and removed the pattern from the Codacy settings, so you needn't do anything.
So that's been resolved.

About the docstring blank line, it seems like there is indeed a conflict between D211 and D203, which I found [here](https://github.com/PyCQA/pydocstyle/issues/141).

I think the other files in the repo are styled in accordance with D211, like [this one](https://github.com/hot9cups/probdists/blob/main/probdists/Generaldistribution.py).

So could you style it similar to that? It should then be throwing a D203 error, and so I'll then add in the 'ignore D203 rule' which should hopefully resolve everything.

**Comment by The-Shwin** [comment]
> 
> 
> @The-Shwin mm, well for the unused import, you could add `# NOQA: F401` to that line which helps resolve the issue.
> However, I've gone ahead and removed the pattern from the Codacy settings, so you needn't do anything.
> So that's been resolved.
> 
> About the docstring blank line, it seems like there is indeed a conflict between D211 and D203, which I found [here](https://github.com/PyCQA/pydocstyle/issues/141).
> 
> I think the other files in the repo are styled in accordance with D211, like [this one](https://github.com/hot9cups/probdists/blob/main/probdists/Generaldistribution.py).
> 
> So could you style it similar to that? It should then be throwing a D203 error, and so I'll then add in the 'ignore D203 rule' which should hopefully resolve everything.
> 
> Also, could you confirm that you were able to run the tests locally and that everything worked as intended?

@hot9cups sure, I've resolved the comments. I was indeed able to run test.py locally, I have a picture of the results here:
image

**Review by hot9cups — APPROVED** [review]
@The-Shwin This looks perfect. Thanks a lot for taking the time and contributing to the library. Is it fine with you if I mention you as a contributor when I release the next version of probdists? :)

**Comment by The-Shwin** [comment]
> 
> 
> @The-Shwin This looks perfect. Thanks a lot for taking the time and contributing to the library. Is it fine with you if I mention you as a contributor when I release the next version of probdists? :)

Thanks! Glad I was able to help. Sure that would be great.

**Pull request merged**